### PR TITLE
Add named frames to CollisionObjects

### PR DESCRIPTION
**Pull request by felixvd**
This belongs to https://github.com/ros-planning/moveit/pull/1060

### CODE REVIEW

File: msg/CollisionObject.msg [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
+# get attached together, as if they are a single object. Objects that do not exist in the world 
+# get removed from the list. Every entry in this list constitutes an attempt to connect this object
+# and the other object in the current world scene.
+string[] connected_objects
+
+# Named frames on the object. Use these to define points of interest on the object that you want
+# to plan with (e.g. screwdriver_tip, kettle_spout, mug_base).
+# The length of these vectors has to be identical.
+string[] frame_names
+geometry_msgs/Pose[] named_frames
```
Comment (davetcoleman) [inline msg/CollisionObject.msg]: rename to frame_poses so it matches frame_names?

File: msg/CollisionObject.msg [hunk 2]
```diff
@@ -1,27 +1,39 @@
...
-# bounding planes (equation is specified, but the plane can be oriented using an additional pose)
+# Bounding planes (equation is specified, but the plane can be oriented using an additional pose)
 shape_msgs/Plane[] planes
 geometry_msgs/Pose[] plane_poses
 
+# List of other objects that this one is rigidly connected to. Objects that are connected move and
+# get attached together, as if they are a single object. Objects that do not exist in the world 
+# get removed from the list. Every entry in this list constitutes an attempt to connect this object
+# and the other object in the current world scene.
+string[] connected_objects
```
Comment (davetcoleman) [inline msg/CollisionObject.msg]: what is the transform from this connected object to others?

File: srv/GetConstraintValidity.srv [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Constraints constraints
```
Comment (davetcoleman) [inline srv/GetConstraintValidity.srv]: add some comments at top for what this service does please

**Comment by felixvd** [comment]
I'm on 16.04 with this system. I can try it out on melodic from November.

I added the changes, but the GetConstraintValidity service might get dropped, as mentioned in the other PR.

**Comment by davetcoleman** [comment]
Did you mean to close this?

**Comment by felixvd** [comment]
I'm not sure what happened, but either way this PR needed to go against melodic, not kinetic. I'm reopening.

**Pull request closed without merging**